### ISSUE TITLE
Batch Training

### ISSUE THREAD

**Issue by HemanthSai7** [VizuaraAI/nano-gpt-oss#2]
Hello, 

Thanks for this awesome repository and the video explaining GPT-OSS. I was going through the repo and have a question regarding training implementation. I think it is not actually using batch training. Please correct my understanding if I'm mistaken.

-  The model is implemented in a way to handle one sample at a time and not a batch
- While going through the training script, I noticed this line.
 ```python
for i in range(len(input_batch)):
        inp = input_batch[i].to(device)
        logits = model(inp)
```
Here we are taking one sample from the batch, calculating the loss and then adding it up

**Comment by amanmurari** [comment]
Yes here i am not using batch training.
